[java][doc] AvoidAssertAsIdentifier and AvoidEnumAsIdentifier - clarify use case

**Affects PMD Version:6.50.0** 

**Description:**

Hi, it is obvious that this rule [AvoidAssertAsIdentifier](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#avoidassertasidentifier) can detect the variable which is named **assert**, but when I run PMD on the code example below, it is crashed unexpectedly. Besides, this case is also shown in the [document](https://pmd.sourceforge.io/pmd-6.50.0/pmd_rules_java_errorprone.html#avoidassertasidentifier).
 
**Exception Stacktrace:**
```
net.sourceforge.pmd.PMDException: Error while parsing /path/to/src/code
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCodeWithoutCache(SourceCodeProcessor.java:124)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:100)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:62)
	at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:89)
	at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:30)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: net.sourceforge.pmd.lang.java.ast.ParseException: Line 3, Column 23: Can't use 'assert' as a variable name when running in JDK 1.4 mode!
	at net.sourceforge.pmd.lang.java.ast.JavaParser.throwParseException(JavaParser.java:30)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.checkForBadAssertUsage(JavaParser.java:35)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.VariableDeclaratorId(JavaParser.java:1999)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.VariableDeclarator(JavaParser.java:1921)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.LocalVariableDeclaration(JavaParser.java:5882)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.BlockStatement(JavaParser.java:5798)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.Block(JavaParser.java:5718)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.MethodDeclaration(JavaParser.java:2148)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.ClassOrInterfaceBodyDeclaration(JavaParser.java:1829)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.ClassOrInterfaceBody(JavaParser.java:1782)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.ClassOrInterfaceDeclaration(JavaParser.java:923)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.TypeDeclaration(JavaParser.java:825)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.CompilationUnit(JavaParser.java:545)
	at net.sourceforge.pmd.lang.java.AbstractJavaParser.parse(AbstractJavaParser.java:62)
	at net.sourceforge.pmd.lang.AbstractParser.doParse(AbstractParser.java:45)
	at net.sourceforge.pmd.SourceCodeProcessor.parse(SourceCodeProcessor.java:136)
	at net.sourceforge.pmd.SourceCodeProcessor.processSource(SourceCodeProcessor.java:200)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCodeWithoutCache(SourceCodeProcessor.java:118)
	... 10 more

```

**Code Sample demonstrating the issue:**

```java
public class C {
    void bar() {
        String assert = "hi";
    }
}
```

**Steps to reproduce:**
Directly run PMD to test this file.

**Running PMD through:** *[Maven]*

--- a/docs/pages/release_notes.md
+++ b/docs/pages/release_notes.md
@@ -15,6 +15,8 @@ This is a {{ site.pmd.release_type }} release.
 ### New and noteworthy
 
 ### Fixed Issues
+* java-errorprone
+    * [#4164](https://github.com/pmd/pmd/issues/4164): \[java]\[doc] AvoidAssertAsIdentifier and AvoidEnumAsIdentifier - clarify use case
 
 ### API Changes
 

--- a/pmd-java/src/main/resources/category/java/errorprone.xml
+++ b/pmd-java/src/main/resources/category/java/errorprone.xml
@@ -130,12 +130,17 @@ public class Violation {
 
     <rule name="AvoidAssertAsIdentifier"
           language="java"
+          maximumLanguageVersion="1.3"
           since="3.4"
           message="Avoid using assert as an identifier; it became a reserved word in JDK 1.4"
           class="net.sourceforge.pmd.lang.rule.XPathRule"
           externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_errorprone.html#avoidassertasidentifier">
         <description>
-Use of the term 'assert' will conflict with newer versions of Java since it is a reserved word.
+Use of the term `assert` will conflict with newer versions of Java since it is a reserved word.
+
+Since Java 1.4, the token `assert` became a reserved word and using it as an identifier will
+result in a compilation failure for Java 1.4 and later. This rule is therefore only useful
+for old Java code before Java 1.4. It can be used to identify problematic code prior to a Java update.
         </description>
         <priority>2</priority>
         <properties>
@@ -341,12 +346,17 @@ private void buz(String x) {}
 
     <rule name="AvoidEnumAsIdentifier"
           language="java"
+          maximumLanguageVersion="1.4"
           since="3.4"
           message="Avoid using enum as an identifier; it's a reserved word in JDK 1.5"
           class="net.sourceforge.pmd.lang.rule.XPathRule"
           externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_errorprone.html#avoidenumasidentifier">
         <description>
-Use of the term 'enum' will conflict with newer versions of Java since it is a reserved word.
+Use of the term `enum` will conflict with newer versions of Java since it is a reserved word.
+
+Since Java 1.5, the token `enum` became a reserved word and using it as an identifier will
+result in a compilation failure for Java 1.5 and later. This rule is therefore only useful
+for old Java code before Java 1.5. It can be used to identify problematic code prior to a Java update.
         </description>
         <priority>2</priority>
         <properties>

